homeassistant/components/lg_thinq/sensor.py

```diff
@@ -0,0 +1,315 @@
+    ),
+    DeviceType.ROBOT_CLEANER: (
+        ERROR_DESC,
+        RUN_STATE_SENSOR_DESC[ThinQProperty.CURRENT_STATE],
+        JOB_MODE_SENSOR_DESC[ThinQProperty.CURRENT_JOB_MODE],
+    ),
+    DeviceType.STICK_CLEANER: (
+        JOB_MODE_SENSOR_DESC[ThinQPropertyEx.CURRENT_JOB_MODE_STICK_CLEANER],
+        RUN_STATE_SENSOR_DESC[ThinQProperty.CURRENT_STATE],
+    ),
+    DeviceType.STYLER: WASHER_SENSORS,
+    DeviceType.WASHCOMBO_MAIN: WASHER_SENSORS,
+    DeviceType.WASHCOMBO_MINI: WASHER_SENSORS,
+    DeviceType.WASHER: WASHER_SENSORS,
+    DeviceType.WASHTOWER_DRYER: WASHER_SENSORS,
+    DeviceType.WASHTOWER: WASHER_SENSORS,
+    DeviceType.WASHTOWER_WASHER: WASHER_SENSORS,
+    DeviceType.WATER_HEATER: (
+        TEMPERATURE_SENSOR_DESC[ThinQProperty.CURRENT_TEMPERATURE],
+    ),
+    DeviceType.WATER_PURIFIER: (
+        RUN_STATE_SENSOR_DESC[ThinQProperty.COCK_STATE],
+        RUN_STATE_SENSOR_DESC[ThinQProperty.STERILIZING_STATE],
+        WATER_INFO_SENSOR_DESC[ThinQProperty.WATER_TYPE],
+    ),
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ThinqConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up an entry for sensor platform."""
+    entities: list[ThinQSensorEntity] = []
+    for coordinator in entry.runtime_data.values():
+        if (
+            descriptions := DEVICE_TYPE_SENSOR_MAP.get(
+                coordinator.api.device.device_type
+            )
+        ) is not None:
+            for description in descriptions:
+                entities.extend(
+                    ThinQSensorEntity(coordinator, description, property_id)
+                    for property_id in coordinator.api.get_active_idx(description.key)
+                )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class ThinQSensorEntity(ThinQEntity, SensorEntity):
+    """Represent a thinq sensor platform."""
+
+    def __init__(
+        self,
+        coordinator: DeviceDataUpdateCoordinator,
+        entity_description: EntityDescription,
+        property_id: str,
+    ) -> None:
+        """Initialize a sensor entity."""
+        super().__init__(coordinator, entity_description, property_id)
+
+        self._attr_options = self.data.options
+        if self.options:
+            self._attr_device_class = SensorDeviceClass.ENUM
+
+    def _update_status(self) -> None:
+        """Update status itself."""
+        super()._update_status()
+
+        _LOGGER.debug(
+            "[%s:%s] update status: %s",
+            self.coordinator.device_name,
+            self.property_id,
+            self.data.value,
+        )
+        self._attr_native_value = self.data.value
+
+        # Update unit.
+        if self.device_class == SensorDeviceClass.ENUM:
+            self._attr_native_unit_of_measurement = None
+            return
+
+        unit_of_measurement = self.data.unit or self.native_unit_of_measurement
```

Comment:
What does the `data.unit` return exactly? When is this set?

Reply:
In temperature, it could be one of Fahreheit or Celsius.